source/views/sis/course-search/search.tsx

```diff
@@ -36,87 +33,110 @@ type ReduxDispatchProps = {
 	loadCourseDataIntoMemory: () => Promise<any>
 }
 
-type Props = ReactProps & ReduxStateProps & ReduxDispatchProps
+type Props = ReduxStateProps & ReduxDispatchProps
 
-type State = {
-	mode: 'loading' | 'pending' | 'ready'
-	isSearchbarActive: boolean
-	typedQuery: string
-}
+type Mode = 'loading' | 'pending' | 'ready'
 
-class CourseSearchView extends React.Component<Props, State> {
-	state = {
-		mode: 'pending',
-		isSearchbarActive: false,
-		typedQuery: '',
+let _debounce = debounce((query: string, callback: () => void) => {
+	if (query.length >= 2) {
+		callback()
 	}
+}, 1500)
+
+const CourseSearchView = (props: Props): JSX.Element => {
+	let [mode, setMode] = React.useState<Mode>('pending')
+	let [typedQuery, setTypedQuery] = React.useState('')
+
+	let navigation = useNavigation()
+
+	React.useLayoutEffect(() => {
+		/**
+		 * SIS top-level recent searches tab has this setup.
+		 *
+		 * When we are one-level deep in a tab navigator, attaching a search bar to the
+		 * view will work fine. See: Recent course search --> Search results
+		 *
+		 * When we are top-level inside a tab navigator, attaching a search bar to the
+		 * view will not work. See: Tab.Navigator > Tab.Screen --> Course search
+		 *
+		 * The `getParent` workaround is provided in the react navigation docs.
+		 * https://reactnavigation.org/docs/navigation-prop#getparent
+		 *
+		 * getParent returns the navigation prop from the parent navigator that the
+		 * current navigator is nested in. If you have a stack navigator and a tab
+		 * navigator nested inside the stack, then you can use getParent inside a screen
+		 * of the tab navigator to get the navigation prop passed from the stack
+		 * navigator.
+		 */
+		navigation.getParent()?.setOptions({
+			headerSearchBarOptions: {
+				barTintColor: c.white,
+				onChangeText: (event: ChangeTextEvent) => {
+					setTypedQuery(event.nativeEvent.text)
+				},
+			},
+		})
+	}, [navigation, typedQuery])
```

Comment:
After testing this I am seeing UI inconsistencies. The searchbar is added to other views in the tab navigator which is not correct.

This could be happening because of how the tab navigator is nested. I am going through the documentation of how to set this up correctly.

- This in particular: https://reactnavigation.org/docs/screen-options-resolution/
- Helpful reference: https://reactnavigation.org/docs/tab-based-navigation

Reply:
Or is it time to break SIS up into multiple top-level buttons?

Tab | Button
---|---
Balances | OleCard
TES | TES
Course Search | SIS
Student Work | … Student Work?

Reply:
Sorry, I meant to say this is **definitely** happening due to `navigation.getParent()?.setOptions()` -- the `.getParent` part being key.

This is our first tab navigation with top-level search in a tab. It works just fine if it's one-level deep, though... meaning Home > SIS tabs > Search > Results, where results has a searchbar.

Breaking these up is still an option, though.

Reply:
I am happy to explore other UI designs, too. How do we feel about a list view?

I think there is a concern about breaking muscle memory on the home screen so if we limited our exploration to the subviews, that could be beneficial?

<img src="https://user-images.githubusercontent.com/5240843/171053399-306e7486-0222-4da3-9791-0e8eeaca8845.png" width=350 />

Reply:
Moved this out into https://github.com/StoDevX/AAO-React-Native/issues/6109 in the interest of solving this in a follow-up.